CMakeLists.txt

```diff
@@ -5,6 +5,24 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
+option(MBED_BUILD_UNITTESTS "Run unit tests only." OFF)
+
+if(MBED_BUILD_UNITTESTS)
+    project(mbed-unittests)
+    enable_testing()
+    
+    add_subdirectory(UNITTESTS)
+    add_subdirectory(${MBED_PATH}/drivers/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/connectivity/cellular/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/connectivity/netsocket/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/connectivity/lorawan/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/storage/blockdevice/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/storage/kvstore/tdbstore/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/storage/kvstore/filesystemstore/tests/UNITTESTS)
+    add_subdirectory(${MBED_PATH}/platform/tests/UNITTESTS)   
+    add_subdirectory(${MBED_PATH}/events/tests/UNITTESTS)
+else()
```

Comment:
We need to keep `else()` to avoid all the target configuration (we are not  generating mbed_build.cmake which is based on target config in targets.json) check like validating target support std vs small library and profile selection check etc.,

Reply:
That seems like a bug in the target validating CMake code. It shouldn't bother checking things that don't matter for the chosen target. Are we checking `CMAKE_CROSSCOMPILING` or `CMAKE_SYSTEM_NAME` to ensure that we don't try to validate things that are invalid or irrelevant for the target platform?

Reply:
We do not but we shall, to check `CMAKE_CROSSCOMPILING`.

Reply:
Used CMAKE_CROSSCOMPILING